ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -540,19 +685,8 @@ void McapHandler::stop_event_thread_nts_()
         event_cv_.notify_one();
         event_thread_.join();
     }
-    std::lock_guard<std::mutex> _(mtx_);
-    clear_all_nts_();
-}
-
-void McapHandler::clear_all_nts_()
-{
-    logInfo(DDSRECORDER_MCAP_HANDLER, "Cleaning all buffers.");
-
-    // Clear samples buffer
     samples_buffer_.clear();
-
-    // Clear pending samples
-    pending_samples_.clear();
+    pending_samples_paused_.clear();
```

Comment:
Pending samples should not be removed when transitioning from PAUSED state to RUNNING, as these are samples that were previously added in RUNNING state (when schema was also not available), and should be kept throughout execution until the schema is received or the handler is stopped.

Clearing pending samples should however be added to the stop function, just in case we might have in the future transitions such as STOPPED -> RUNNING -> STOPPED -> RUNNING